reactor-core/src/main/java/reactor/core/publisher/FluxTimeout.java

```diff
@@ -87,13 +99,19 @@ public void subscribe(CoreSubscriber<? super T> actual) {
 		source.subscribe(main);
 	}
 
+	private String resolveName(Publisher<?> source) {
+		Tuple2<String, List<Tag>> nameAndTags = FluxMetrics.resolveNameAndTags(source);
+		return nameAndTags.getT1();
+	}
+
 	static final class TimeoutMainSubscriber<T, V>
 			extends Operators.MultiSubscriptionSubscriber<T, T> {
 
 		final Function<? super T, ? extends Publisher<V>> itemTimeout;
 
 		final Publisher<? extends T> other;
 		final String timeoutDescription; //only useful/non-null when no `other`
+		final String name;
```

Comment:
other option: make `name` `@Nullable`...

Reply:
we shouldn't even need a name, just reusing timeoutDescription should be fine. Also only doing that if there is no other.